### PR TITLE
feat: parse script of records with a given name

### PR DESCRIPTION
**Pull request by RustomMS**

### CODE REVIEW

**Comment by skyzh** [comment]
Would you please also update the existing runner to use this new interface? Thanks a lot!

**Comment by RustomMS** [comment]
@skyzh For sure, I am assuming the new commit is what you intended

File: sqllogictest/src/runner.rs [hunk 1]
```diff
@@ -802,6 +802,16 @@ impl<D: AsyncDB> Runner<D> {
...
+        script: &str,
+        name: impl Into<Arc<str>>,
+    ) -> Result<(), TestError> {
+        let records = parse_with_name(script, name).expect("failed to parse sqllogictest");
+        self.run_multi_async(records).await
+    }
+
     /// Run a sqllogictest file.
     pub async fn run_file_async(&mut self, filename: impl AsRef<Path>) -> Result<(), TestError> {
         let records = parse_file(filename)?;
```
Comment (skyzh) [inline sqllogictest/src/runner.rs:817]: I think you can also parse name here? Thanks for fixing! I'll run a local test soon and get this merged.

Comment (RustomMS) [reply sqllogictest/src/runner.rs:817]: I believe `parse_file` includes the `filename` as the `file` field in the `Location` provided to `parse_file_inner`. So we should be all good for `run_file*` functions

See:
https://github.com/risinglightdb/sqllogictest-rs/blob/f7c12f79436a8384f952f2652d43c8d2978fb4e4/sqllogictest/src/parser.rs#L593-L597

**Pull request merged**